zilencer/views.py

```diff
@@ -216,13 +220,73 @@ def unregister_all_remote_push_devices(
     return json_success(request)
 
 
+def delete_duplicate_registrations(
+    registrations: List[RemotePushDeviceToken], server_id: int, user_id: int, user_uuid: str
+) -> List[RemotePushDeviceToken]:
+    """
+    When migrating to support registration by UUID, we introduced a bug where duplicate
+    registrations for the same device+user could be created - one by user_id and one by
+    user_uuid. Given no good way of detecting these duplicates at database level, we need to
+    take advantage of the fact that when a remote server sends a push notification request
+    to us, it sends both user_id and user_uuid of the user.
+
+    This function, knowing the user_id and user_uuid of the user, can detect duplicates
+    and delete the legacy user_id registration if appropriate.
+
+    Return the list of registrations with the user_id-based duplicates removed.
+    """
+
+    # All registrations passed here should be of the same kind (apple vs android).
+    assert len({registration.kind for registration in registrations}) == 1
+    kind = registrations[0].kind
+
+    tokens = [device.token for device in registrations]
+    tokens_counter = Counter(tokens)
```

Comment:
nit:
```suggestion
    tokens_counter = Counter(device.token for device in registrations)
```

AFAICS `tokens` isn't used again, so we can skip allocating the list and have `Counter` just consume a generator expression.